types.go

```diff
@@ -83,3 +61,70 @@ func (m PubKey) AsABCI() abci.PubKey {
 		Type: m.Type,
 	}
 }
+
+func (m ValidatorUpdates) Validate() error {
+	var err error
+	for _, v := range m.ValidatorUpdates {
+		err = errors.Append(err, v.Validate())
+	}
+	return err
+}
+
+// Deduplicate makes sure we only use the last validator update to any given validator.
+// For bookkeeping we have an option to drop validators with zero power, because those
+// are being remove by tendermint once propagated.
+func (m ValidatorUpdates) Deduplicate(dropZeroPower bool) []ValidatorUpdate {
+	duplicates := make(map[string]int, 0)
```

Comment:
I think it's better not to force the map to have a zero size. Skipping this argument and using default might be better.

Reply:
Yeah, it does not need to be done.